fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -468,13 +472,25 @@ public Response patch(@PathParam("type") String type, @PathParam("id") String id
             response = addHeaders(response, resource);
             return response.build();
         } catch (FHIRPersistenceResourceNotFoundException e) {
-            return exceptionResponse(e, Response.Status.METHOD_NOT_ALLOWED);
+            status = Status.METHOD_NOT_ALLOWED;
+            return exceptionResponse(e, status);
         } catch (FHIROperationException e) {
-            return exceptionResponse(e);
+            status = issueListToStatus(e.getIssues());
+            return exceptionResponse(e, status);
         } catch (Exception e) {
-            return exceptionResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+            status = Status.INTERNAL_SERVER_ERROR;
+            return exceptionResponse(e, status);
         } finally {
-            log.exiting(this.getClass().getName(), "patch(String,String,JsonArray)");
+            try {
+                RestAuditLogger.logPatch(httpServletRequest,
```

Comment:
repeated pattern... seems like an enum and a single method may be best.

Reply:
will not refactor this for this PR...it can get revisited later